### PR TITLE
[W6.4d][W13-B2]Raynold Ng

### PR DESCRIPTION
**Pull request by raynoldng**
LO-Polymorphism

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -70,4 +74,8 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating() {
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:78]: Need precise header comment to indicate what this method is for.

File: src/seedu/addressbook/commands/ExitCommand.java [hunk 2]
```diff
@@ -11,6 +11,10 @@
             + "Example: " + COMMAND_WORD;
     public static final String MESSAGE_EXIT_ACKNOWEDGEMENT = "Exiting Address Book as requested ...";
 
+    public ExitCommand() {
+        super(true);
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/ExitCommand.java:15]: does exit mutate data?

File: src/seedu/addressbook/logic/Logic.java [hunk 3]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java:88]: need to update the header comment to indicate the new behavior of save.

**Review by okkhoy — COMMENTED** [review]
This is a different approach I haven't seen in quite some time :+1:

**Comment by okkhoy** [comment]
@raynoldng 
Some comments added. Please ack & close the PR after reading comments.

**Comment by raynoldng** [comment]
Acknowledged with thanks

**Pull request closed without merging**